README.md

```diff
@@ -1,4 +1,201 @@
+root of the repository) you can use [a2z.py](a2z.py) to generate an empty one. You
+can either provide the repository name directly:
+
+```bash
+$ a2z init singularityhub/sregistry
+INFO:a2z:Generating .all-contributorsrc
+```
+
+or export it to the environment as `GITHUB_REPOSITORY` (this is how it would be discovered
+during a GitHub workflow run).
+
+```bash
+export GITHUB_REPOSITORY=singularityhub/sregistry
+$ a2z init
+INFO:a2z:Generating .all-contributorsrc
+```
+In both cases, you'll generate an empty file:
+
+```bash
+$ cat .all-contributorsrc 
+{
+    "projectName": "sregistry",
+    "projectOwner": "singularityhub",
+    "repoType": "github",
+    "repoHost": "https://github.com",
+    "files": [
+        "README.md"
+    ],
+    "imageSize": 100,
+    "commit": true,
+    "commitConvention": "none",
+    "contributors": [],
+    "contributorsPerLine": 7
+}
+```
+
+### 2. Generate .zenodo.json
+
+If you want to generate a fresh Zenodo.json, you can do that as follows:
+
+```bash
+$ a2z init --zenodo 10.5281/zenodo.1012531 singularityhub/sregistry
+```
+
+While the GitHub repository is sometimes found as a related identifier,
+this isn't always the case, so you should again provide it on the command
+line or exported to `GITHUB_REPOSITORY`. By default, we parse contributors
+from the GitHub API, and include the creators already defined in Zenodo.
+You will need this `.zenodo.json` file to exist in order to update it.
+
+### 3. Update contributors
+
+We next want to use the GitHub api to discover contributors to the repository,
+for either both the `.all-contributorsrc` and the `.zenodo.json`, or just one of the two.
+
+#### Update all
+
+If you want to update both the allcontributors file and the zenodo.json, you
+can kill two birds with one stone (and essentially cache the GitHub API request)
+and do the following:
+
+```bash
+$ a2z update all
+```
+
+Note that for the update, we query the GitHub API to update both the zenodo.json
+and contributors file.
+
+#### Update allcontributors
+
+Since we already have the contributors file we don't need to provide the repository name again, however it's suggested that you export a `GITHUB_TOKEN` to increase your API limits (if necessary).
+
+```bash
+$ a2z update allcontrib
+```
+
+You can optionally set a minimum number of contributors threshold to add (defaults to 1), or
+a string to represent the kind of contribution (defaults to "core")
+
+```bash
+$ a2z update --thresh 10 --ctype docs
+```
+
+Note that GitHub bots are not included as contributors, and they are indicated with
+"[bot]" in the name. If you find that you hit the API limit, then you will see this:
+
+```bash
+$ a2z update allcontrib
+INFO:a2z:Updating .all-contributorsrc
+Response 403: rate limit exceeded, cannot retrieve user RonaldEnsing.
+```
+And should export a `GITHUB_TOKEN` to increase it.
+
+#### Update zenodo.json
+
+For the zenodo update alone, since we don't read a repository from the .all-contributors file,
```

Comment:
but why don't we?  it is after all "allcontributors2zenodo" ;)

Reply:
Because that's a dependency that I don't want to add.

Reply:
but if I do have `.all-contributorsrc` already -- why it should not be used???  that was the original goal